### PR TITLE
[BUGFIX] fix NavDrawer to force href update of menu items

### PR DESCRIPTION
**Pull request by romanwozniak**
In https://github.com/gojek/mlp/pull/30, `NavDrawer` component was updated to use `EuiTreeView` instead of `EuiListGroup`. `EuiTreeView` doesn't support `href` property, so menu navigation was implemented with `onClick` property and then `navigate` to the desired destination. 

However, this doesn't work properly, because mlp applications are deployed as separate single page apps, so the destination utl could be outside of the application's state (i.e. if user clicks on `Merlin` menu item from the `Turing` app, then this page doesn't exist in Turing's router). To solve this, the page should be forcefully refreshed and this is achieved with update of `window.location.href`.

On top of this, I've updated dependencies and fixed some of rollup warnings (such as circular dependency).

### CODE REVIEW

File: ui/packages/lib/src/components/header/Header.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ import { HeaderUserMenu } from "./HeaderUserMenu";
 import { slugify } from "../../utils";
 import { NavDrawer } from "../nav_drawer";
 import "./Header.scss";
+import {Link} from "@reach/router";
```
Comment (terryyylim) [inline ui/packages/lib/src/components/header/Header.js:18]: super nit: couple of the same thing, to add spaces around curly braces

**Review by terryyylim — APPROVED** [review]

File: ui/packages/lib/src/components/nav_drawer/NavDrawer.js [hunk 2]
```diff
@@ -53,7 +58,9 @@ export const NavDrawer = ({ homeUrl = "/", docLinks }) => {
 
         callback: () =>
           !children || !projectId
-            ? navigate(projectId ? `${a.href}/projects/${projectId}` : a.href)
+            ? window.location.href = !!projectId
```
Comment (ariefrahmansyah) [inline ui/packages/lib/src/components/nav_drawer/NavDrawer.js:61]: I'm curious about this line. At first, I thought it should be a condition expression for the ternary operator, but it seems you are assigning value to `window.location.href` by using one `=`. Is it another JS magic that I don't know? Please enlighten me.

File: ui/packages/lib/src/components/header/Header.js (hunk 1 above)
Comment (romanwozniak) [reply ui/packages/lib/src/components/header/Header.js]: fixed and also updated `husky` to automatically format files according prettier configuration on pre-commit hook

File: ui/packages/lib/src/components/nav_drawer/NavDrawer.js (hunk 2 above)
Comment (romanwozniak) [reply ui/packages/lib/src/components/nav_drawer/NavDrawer.js]: Which part is magical here? :) yeah, I'm assigning a value (conditional) to `window.location.href` and with that you can sort of programatically trigger a click on a link with assigned href value. 
Trick is that `navigate` comes from `@reach/router` and it navigates inside a single page app (SPA) (without triggering a full refresh of the page), so I need to emulate a click on a link to completely refresh the page (so another SPA can be downloaded)

**Pull request merged**